File: 20171018-Seattle/wrapped_structptr.c
struct HmdMatrix44_t system_GetProjectionMatrix(struct VR_IVRSystem_FnTable* iSystem, EVREye eEye, float fNearZ, float fFarZ) {
    return iSystem->GetProjectionMatrix(eEye, fNearZ, fFarZ);
}

struct HmdMatrix34_t system_GetEyeToHeadTransform(struct VR_IVRSystem_FnTable* iSystem, EVREye eEye) {
    return iSystem->GetEyeToHeadTransform(eEye);
}

bool system_ComputeDistortion(struct VR_IVRSystem_FnTable* iSystem, EVREye eEye, float fU, float fV, struct DistortionCoordinates_t* dest) {
    return iSystem->ComputeDistortion(eEye, fU, fV, dest) != 0;
}

bool system_IsTrackedDeviceConnected(struct VR_IVRSystem_FnTable* iSystem, TrackedDeviceIndex_t unDeviceIndex) {
    return iSystem->IsTrackedDeviceConnected(unDeviceIndex);
}

/* ... */

File: 20171018-Seattle/wrapped_structptr.go
type System struct {
	ptr *C.struct_VR_IVRSystem_FnTable
}

func (sys *System) GetProjectionMatrix(eye int, near, far float32, dest *mgl.Mat4) {
	m44 := C.system_GetProjectionMatrix(sys.ptr, C.EVREye(eye), C.float(near), C.float(far))
	/* reorder the matrix for compatability with OpenGL */
}

func (sys *System) GetEyeToHeadTransform(eye int, dest *mgl.Mat3x4) {
	m34 := C.system_GetEyeToHeadTransform(sys.ptr, C.EVREye(eye))
	/* reorder the matrix for compatability with OpenGL */
}

/* ... */